hil/ext/switches/dellnos9.py

```diff
@@ -0,0 +1,387 @@
+
+    def _port_on(self, interface):
+        """ Turns on <interface>
+
+        Turn on port and enable hybrid portmode and switchport.
+        Fast method that uses the REST API (no hackery).
+        """
+
+        url = self._construct_url(interface=interface)
+        interface = self._convert_interface(self.interface_type) + \
+            interface.replace('/', '-')
+        payload = '<interface><name>%s</name><portmode><hybrid>true' \
+                  '</hybrid></portmode><switchport></switchport>' \
+                  '<shutdown>false</shutdown></interface>' % interface
+
+        self._make_request('PUT', url, data=payload)
+
+    def _is_port_on(self, port):
+        """ Returns a boolean that tells the status of a switchport
+        Fast method that uses the REST API (no hackery)
+        """
+
+        url = self._construct_url(interface=port, suffix='\?with-defaults')
+        response = self._make_request('GET', url)
+        root = etree.fromstring(response.text)
+        shutdown = root.find(self._construct_tag('shutdown')).text
+        # is error handling needed here?
+        return (shutdown == 'false')
+
+    # HELPER METHODS *********************************************
+    def _construct_url(self, interface=None, suffix=''):
+        """ Construct the API url for a specific interface appending suffix.
+
+        Args:
+            interface: interface to construct the url for
+            suffix: suffix to append at the end of the url (for get methods)
+
+        Returns: string with the url for a specific interface and operation
+
+        If interface is None, then it returns the URL for REST API CLI.
+        """
+
+        if interface is None:
+            return '%s/api/running/dell/_operations/cli' % self.hostname
+
+        val = re.compile(r'^\d+/\d+(/\d+)?$')
+        if val.match(interface):
+            # if `interface` refers to port name
+            # the urls have dashes instead of slashes in interface names
+            interface = interface.replace('/', '-')
+            interface_type = self._convert_interface(self.interface_type)
+        else:
+            # interface refers to `vlan`
+            interface_type = 'vlan-'
+
+        return '%(hostname)s/api/running/dell/interfaces/interface/' \
+            '%(interface_type)s%(interface)s%(suffix)s' \
+            % {
+                'hostname': self.hostname,
+                'interface_type': interface_type,
+                'interface': interface,
+                'suffix': '/%s' % suffix if suffix else ''
+            }
+
+    @staticmethod
+    def _convert_interface(interface_type):
+        """ Convert the interface name from switch CLI form to what the API
+        server understands.
+
+        Args:
+            interface: the interface in the CLI-form
+
+        Returns: string interface
+        """
+        iftypes = {'GigabitEthernet': 'gige-', 'TenGigabitEthernet': 'tengig-',
```

Comment:
Keep these to one key-value pair per line; having two on some of the lines makes it much harder to read.